cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/profiles_populator.py

```diff
@@ -0,0 +1,505 @@
+        return True  # it's ok to change mind
+
+    # Login to Azure account
+    cli_ask("Please login to target Azure account as a privileged user [ENTER]")
+    account = azure_login_interactively()
+    if account is None:
+        log.error("Failed to login into Azure account")
+        return False
+
+    # Get or create Service Principal
+    principal = get_service_principal(SP_NAME, profiles) or setup_service_principal(account.subscription_id, SP_NAME)
+    if principal is None:
+        log.error("Failed to get as well as to create Service Principal in subscription '%s'", account.subscription_id)
+        return False
+
+    # Request Azure location and list Resource Groups in that location
+    location = cli_ask_azure_location()
+    resource_group_names = list_resource_groups(location)
+
+    # Update profile list with a new item
+    profile = AzureProfile(
+        name=profile_name,
+        subscription_id=account.subscription_id,
+        tenant_id=account.tenant_id,
+        principal_id=principal.id,
+        principal_secret=principal.secret,
+        location=location,
+        resource_group_names=resource_group_names,
+        storage_account_names=[],
+    )
+    insert_or_overwrite_profile(profiles, profile)
+
+    cli_tell(f"Profile '{profile_name}' ready")
+    return True
+
+
+def cli_ask_profile_name() -> str:
+    while True:
+        name = cli_ask("Enter name for a new profile: ")
+        if name != "" and " " not in name and "\t" not in name:
+            return name
+
+        cli_tell("Name must not contain white space characters")
+
+
+def cli_ask_overwrite_profile(profile_name: str) -> bool:
+    answer = cli_ask(f"Profile '{profile_name}' already exists. Overwrite? [y/n]: ")
+    return answer.lower() == "y"
+
+
+def cli_ask_azure_location() -> str:
+    NUM_COLUMNS_FOR_LOCATIONS_PRINTOUT = 3
+    locations = list_locations()
+    num_locations = len(locations)
+    if num_locations == 0:
+        log.warning("No locations are available; skipping location selection")
+        return ""
+
+    cli_tell("Select Azure Location. Available locations: ")
+    cli_tell(format_columns(locations, NUM_COLUMNS_FOR_LOCATIONS_PRINTOUT))
+    selected_index = cli_ask_number_in_range(num_locations)
+    return locations[selected_index]
+
+
+def cli_ask_number_in_range(numbers_range: int) -> int:
+    while True:
+        s = cli_ask(f"Enter number in range [{0} - {numbers_range-1}]: ")
+        if not s.isdigit():
+            cli_tell("Invalid number")
+            continue
+        number = int(s)
+        if number not in range(numbers_range):
+            cli_tell("Number out of range")
+            continue
+        return number
+
+
+def cli_ask(prompt: str) -> str:
+    """Ask user for information and return the answer"""
+
+    answer = input(prompt).strip()
+    return answer
+
+
+def cli_tell(msg: str = "") -> None:
```

Comment:
It complements the whole cli_* function family. Can eg. add text coloring here and it may actually be a good idea given that the user interaction is interleaving with logger output